### PR TITLE
fix(build): Dependency cycle in packages

### PR DESCRIPTION
**Pull request by jtaaa**
This PR is v ⚓
It includes a bunch of stuff that maybe should have been split into separate PRs

I recommend just taking a quick skim and we can ensure everything works through Travis.

I went ahead and did the storybooks setup for `shared-components` in this PR for simplicity instead of tryna merge your PR @chriskrogh but it's mostly the same.

## What's changed

1. Added the `shared-components` package and moved all shared components from `shared`
2. Setup storybooks in `shared-components` package and moved all shared component stories
3. Removed and ignored all generated files in Git
4. Added the `generate` package script to `server` and `root` packages

## Notes

Running the new `yarn generate` script root directory will generate the `schema.gql` file is required by the `seller-client` and `marketplace-client` applications. It is also added as the first step of the `yarn build:shared-components` script which is run by `yarn build:seller` and `yarn build:server` so this shouldn't require changes to our cicd setup.

### CODE REVIEW

**Comment by jtaaa** [comment]
Task linked: [CU-cetfrc Stop tracking generated files](https://app.clickup.com/t/cetfrc)

File: packages/server/src/utils/apollo/schema.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      PickupResolver,
+      DeliveryResolver,
+      ProductResolver,
+      ProductOptionResolver,
+      TagResolver,
+      SearchResolver,
+      OrderResolver,
+      ImageResolver,
+    ],
+    emitSchemaFile: true,
```
Comment (jtaaa) [inline packages/server/src/utils/apollo/schema.ts:31]: Automatically emit in `process.cwd()` dir with `schema.gql` name

File: packages/shared-components/.storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export const parameters = {
+  actions: { argTypesRegex: '^on[A-Z].*' },
+};
+
+addDecorator((storyFn) => (
+  <ThemeProvider theme={theme}>
+    <CssBaseline />
+    {storyFn()}
+  </ThemeProvider>
+));
```
Comment (jtaaa) [inline packages/shared-components/.storybook/preview.js:16]: Use theme in storybooks. Must import React to use JSX

**Comment by jtaaa** [comment]
it builds! 🔧

File: packages/shared-components/.storybook/preview.js (hunk 2 above)
Comment (chriskrogh) [reply packages/shared-components/.storybook/preview.js:16]: Ahh I was missing this when I was playing around

**Review by chriskrogh — APPROVED** [review]

**Comment by chriskrogh** [comment]
Does this mean that I can close https://github.com/SYSTT/sys-monorepo/pull/133/files ?

**Comment by jtaaa** [comment]
> Does this mean that I can close #133 ?

yessir! I'll do it here

closes #133

**Pull request merged**